slsDetectorServers/slsDetectorServer/src/slsDetectorServer_funcs.c

```diff
@@ -6332,16 +6332,14 @@ int set_veto_photon(int file_des) {
     int args[2] = {-1, -1};
     if (receiveData(file_des, args, sizeof(args), INT32) < 0)
         return printSocketReadError();
-    int chipIndex = args[0];
-    int numChannels = args[1];
+    const int chipIndex = args[0];
+    const int numChannels = args[1];
 
-    int gainIndices[args[1]];
-    memset(gainIndices, 0, sizeof(gainIndices));
+    int gainIndices[numChannels];
     if (receiveData(file_des, gainIndices, sizeof(gainIndices), INT32) < 0)
         return printSocketReadError();
 
-    int values[args[1]];
-    memset(values, 0, sizeof(values));
+    int values[numChannels];
```

Comment:
In this case you will still get a VLA, since numChannels is not known at compile time. 
It's only in the case below that you avoid a VLA
```cpp
const int size = 5; //this is known at compile time
int arr[size]; // now the array size is fixed
```

But since VLAs are used a quite a bit in the servers we don't need to address all the cases in this PR.

Reply:
alright. also because we have to use VLAs in the server as we dont know the size for many of them at compile time.